Add non-failure status message to plugin checker

ianpatt opened the pull request:
SKSE 2.1.3 adds some new error reporting for plugins. Most of it was designed to be backwards compatible, but a non-failure status is now reported in the log for DLLs that don't look like SKSE plugins (can happen in legit cases for dependent DLLs).

Example line: `plugin avcodec-58.dll (00000000  00000000) no version data 0 (handle 0)`

LostDragonist commented:
@ianpatt Any chance you can give me an example skse64.log to test this out with?  I have a feeling that "0" between "no version data" and the handle is gonna cause an issue with the == you added.

Holt59 commented:
> @ianpatt Any chance you can give me an example skse64.log to test this out with? I have a feeling that "0" between "no version data" and the handle is gonna cause an issue with the == you added.

Yes, that probably won't work since the regex will match the `0` with load status, so `__loadStatus` will contain `"no version data 0"`.

ianpatt commented:
Here's an excerpt with a few different types of errors/success.

```
checking plugin avcodec-58.dll
plugin avcodec-58.dll (00000000  00000000) no version data 0 (handle 0)
checking plugin testdll - badversion.dll
plugin testdll - badversion.dll (00000001 bad version plugin 00000001) disabled, incompatible with current runtime version 0 (handle 0)
checking plugin testdll - versiondb.dll
checking plugin testdll.dll
plugin testdll.dll (00000001 my awesome plugin 00000001) disabled, incompatible with current runtime version 0 (handle 0)
loading plugin "needs new version database plugin"
plugin testdll - versiondb.dll (00000001 needs new version database plugin 00000001) loaded correctly (handle 1)
```

It looks like two modifications need to be made to the regexp. First, the name field can be empty (changed to * instead of +). Second, I added a potential error code to the log, but forgot to add it to the success case. Added optional `(?P<errorCode> \d+)?` to capture that data. This matches my test cases, but feedback from people with more regexp best practice would be appreciated.

LostDragonist commented:
Looks good from what testing I managed to cobble up.  Thanks @ianpatt!
